src/crimson/common/tri_mutex.h

```diff
@@ -3,9 +3,15 @@
 
 #pragma once
 
+#include <concepts>
+
 #include <seastar/core/future.hh>
 #include <seastar/core/circular_buffer.hh>
 
+template <typename Lock>
+concept TriMutexSyncLock = std::is_void_v<
+  std::invoke_result_t<decltype(&Lock::lock), Lock> >;
```

Comment:
if we have to add a concept for [BasicLockable](http://www.cplusplus.com/reference/concept/BasicLockable/). might want to put something like:
```c++
template<typename Mutex>
concept basic_lockable = requires(Mutex m) {
    { m.lock()   } -> std::same_as<void>;
    { m.unlock() } -> std::same_as<void>;
};
```

Reply:
Discarded the concept.